src/core/lib/security/transport/secure_endpoint.cc

```diff
@@ -410,8 +411,26 @@ static void endpoint_write(grpc_endpoint* secure_ep, grpc_slice_buffer* slices,
 
     if (ep->zero_copy_protector != nullptr) {
       // Use zero-copy grpc protector to protect.
-      result = tsi_zero_copy_grpc_protector_protect(ep->zero_copy_protector,
-                                                    slices, &ep->output_buffer);
+      grpc_slice_buffer_reset_and_unref_internal(
```

Comment:
What is the reason to  reset / unref at the start of the if statement instead of at the end?

Reply:
Good catch. Moved it to the end of the if statement.